### PR TITLE
Feature/switch to fhir valuesets

### PR DESCRIPTION
**Pull request by BjoernKrollGba**
I also integrated lombok. When using lombok it is helpful to install a lombok plugin in the IDE (e.g. IntelliJ). Please, tell me your opinion about using lombok. I could remove it and readd the boilerplate code of getters and setters.

I need not adopt the Samply Parent POM to use Spring Boot. Instead I added the Spring Boot dependency management and the Spring Boot Maven Plugin.

### CODE REVIEW

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -20,98 +20,99 @@
...
+   * Name of the property which should be used to override the default download url for the wrapper.
+   */
+  private static final String PROPERTY_NAME_WRAPPER_URL = "wrapperUrl";
 
-    public static void main(String args[]) {
-        System.out.println("- Downloader started");
-        File baseDirectory = new File(args[0]);
-        System.out.println("- Using base directory: " + baseDirectory.getAbsolutePath());
+  public static void main(String args[]) {
+    System.out.println("- Downloader started");
```
Comment (TasDeniz) [inline .mvn/wrapper/MavenWrapperDownloader.java:48]: Maybe use Logger instead of System.out.println?

File: pom.xml [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
     <properties>
         <java.version>11</java.version>
     </properties>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-dependencies</artifactId>
+                <version>2.4.0</version>
```
Comment (TasDeniz) [inline pom.xml:25]: add the version to the properties

**Review by TasDeniz — CHANGES_REQUESTED** [review]
Small changes needed
Added comments to the pr

File: .mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (BjoernKrollGba) [reply .mvn/wrapper/MavenWrapperDownloader.java:48]: Da hast Du im Prinzip Recht. Allerdings möchte ich den MavenWrapperDownloader.java nicht unbedingt anfassen, da er mir generiert wurde. Das einzige was ich ändern musste, ist die Formatierung.

File: pom.xml (hunk 2 above)
Comment (BjoernKrollGba) [reply pom.xml:25]: @alexanderkiel Wie ist der Stand nochmal, hatten wir uns darauf geeinigt i.d.R. keine Properties für Versionen anzulegen? Ich bin mir nicht mehr ganz sicher.

Comment (alexanderkiel) [reply pom.xml:25]: Yes we agreed to not use properties for single versions that can be just written inline. I would still use properties for versions that are used more than once in order to synchronize one version over multiple dependencies.

File: src/main/java/de/samply/icd10dictionary/model/IcdCode.java [hunk 3]
```diff
@@ -1,68 +1,16 @@
...
-    }
-
-    public String getKind() {
-        return kind;
-    }
-
-    public void setKind(String kind) {
-        this.kind = kind;
-    }
+import lombok.AllArgsConstructor;
```
Comment (alexanderkiel) [inline src/main/java/de/samply/icd10dictionary/model/IcdCode.java:3]: Why don't we just use the HAPI Model Classes?

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

File: .mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (alexanderkiel) [reply .mvn/wrapper/MavenWrapperDownloader.java:48]: Why do we need this wrapper?

File: pom.xml [hunk 4]
```diff
@@ -90,10 +107,40 @@
     </dependencies>
 
     <build>
+        <finalName>icd10-dictionary</finalName>
```
[the remark this answers is not in the document]
Comment (BjoernKrollGba) [reply pom.xml:110]: The artefactId contains the version. However, the Docker relies on the fixed name "icd10-dictionary".

File: src/main/java/de/samply/icd10dictionary/model/IcdCode.java (hunk 3 above)
Comment (BjoernKrollGba) [reply src/main/java/de/samply/icd10dictionary/model/IcdCode.java:3]: To keep this prject slim and avoid unnecessary dependencies. But you can write an issue and we can prioratize it.

File: .mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (alexanderkiel) [reply .mvn/wrapper/MavenWrapperDownloader.java:48]: I see. Maven Wrapper is a thing if you don't have Maven installed. Do you use the mvnw command to build the project? If I remove the `.mvn` directory and use just `mvn clean install`, it works without creating `.mvn` again. I would strongly suggest to remove the wrapper stuff. How is it part of string-boot? Does the project generator generate the project this way?

File: pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply pom.xml:113]: Ok, I see.

File: src/main/java/de/samply/icd10dictionary/model/IcdCode.java (hunk 3 above)
Comment (alexanderkiel) [reply src/main/java/de/samply/icd10dictionary/model/IcdCode.java:3]: Ok lombok is really nice. Didn't know it. So may be even better than use the overly blown HAPI Model Classes.

**Review by alexanderkiel — CHANGES_REQUESTED** [review]
- remove maven wrapper
- answer questions regarding URL machting and whitespace
- document usage of lombok in README

**Comment by BjoernKrollGba** [comment]
> * remove maven wrapper
DONE
> * answer questions regarding URL machting and whitespace
CHANGED CODE
> * document usage of lombok in README
ADDED
> * add version to lombok in POM
The version is managed by Spring Boot.

**Review by alexanderkiel — CHANGES_REQUESTED** [review]
Can you please also remove the `mvnw` and `mvnw.cmd` files?

File: pom.xml [hunk 5]
```diff
@@ -1,23 +1,34 @@
...
-        <relativePath/> <!-- lookup parent from repository -->
-    </parent>
-    <groupId>de.samply</groupId>
     <artifactId>icd10-dictionary</artifactId>
-    <version>0.0.1-SNAPSHOT</version>
+    <version>1.0.0-SNAPSHOT</version>
     <name>icd10-dictionary</name>
     <description>REST Service for ICD10 lookups</description>
 
+    <parent>
```
[the remark this answers is not in the document]
Comment (TasDeniz) [reply pom.xml:6]: If you want to upload your artifact to maven central:
Please add inside of the pom.xml the informations which are required for maven central. -> https://central.sonatype.org/pages/requirements.html#sufficient-metadata
Pom.xml convention can be found here:
https://maven.apache.org/developers/conventions/code.html#pom-code-convention

**Review by TasDeniz — CHANGES_REQUESTED** [review]
More pom informations needed for maven central

**Review by alexanderkiel — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -14,6 +14,23 @@
...
     <description>REST Service for ICD10 lookups</description>
+    <url>https://github.com/samply/maven-parent/icd10-dictionary</url>
+
+    <licenses>
+        <license>
+            <name>The Apache Software License, Version 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+        </license>
+    </licenses>
+
```
Comment (TasDeniz) [inline pom.xml:25]: You have forgotten the <scm> and <organisation> tag.
You can find a full example in the pom.xml of the share-client. https://github.com/samply/share-client/blob/master/pom.xml

**Review by TasDeniz — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 6 above)
Comment (BjoernKrollGba) [reply pom.xml:25]: Why do I have to explicitly state these two tags? They are inherited by the Parent-POM (part of effective POM):
`
  <organization>
    <name>Samply Community</name>
  </organization>

...

  <scm>
    <connection>scm:git:https://github.com/samply/maven-parent.git/icd10-dictionary</connection>
    <developerConnection>scm:git:https://github.com/samply/maven-parent.git/icd10-dictionary</developerConnection>
    <url>https://github.com/samply/maven-parent/icd10-dictionary</url>
  </scm>
`

**Review by TasDeniz — APPROVED** [review]

**Pull request merged**